driver/src/main/java/org/neo4j/driver/Values.java

```diff
@@ -386,6 +395,98 @@ public static Value value(Stream<Object> stream) {
         return new ListValue(values);
     }
 
+    /**
+     * Returns a {@link TypeSystem#MAP() map} value based on {@link java.lang.reflect.RecordComponent record components}
+     * of a given {@link java.lang.Record Java Record}.
+     * <p>
+     * Example (similar to the <a href=https://github.com/neo4j-graph-examples/movies>Neo4j Movies Database</a>):
+     * <pre>
+     * {@code
+     * // assuming the following Java record
+     * public record Movie(String title, String tagline, long released) {}
+     * // a new movie may be created in the following way
+     * var movie = new Movie("title", "tagline", 2025);
+     * var movieValue = Values.value(movie);
+     * driver.executableQuery("CREATE (:Movie $movie)")
+     *         .withParameters(Map.of("movie", movieValue))
+     *         .execute();
+     * }
+     * </pre>
+     * Because the driver methods accepting a {@code Map<String, Object>} as query parameters automatically map values
+     * to {@link Value} instances, it is possible to avoid mapping movie explicitly:
+     * <pre>
+     * {@code
+     * var movie = new Movie("title", "tagline", 2025);
+     * driver.executableQuery("CREATE (:Movie $movie)")
+     *         .withParameters(Map.of("movie", movie))
+     *         .execute();
+     * }
+     * </pre>
+     * Assuming movie titles being unique, it is possible to update the created movie in the following way:
+     * <pre>
+     * {@code
+     * var updatedMovie = new Movie("title", "updated tagline", 2024);
+     * driver.executableQuery("""
+     *                 MATCH (movie:Movie {title: $movie.title})
+     *                 SET movie += $movie
+     *                 """)
+     *         .withParameters(Map.of("movie", updatedMovie))
+     *         .execute();
+     * }
+     * </pre>
+     * The {@link Property} annotation may be used to override the record component name.
+     * <pre>
+     * {@code
+     * public record Movie(String title, String tagline, @Property("releasedYear") long released) {}
+     * }
+     * </pre>
+     * Note that those record components that have {@code null} value will be excluded from the map value.
+     * <p>
+     * It is also important to understand that sending all properties over network may not always be desirable and will
+     * depend on a use-case.
+     * <p>
+     * In addition, please note that while this mapping may build nested structures, like map of maps, there may be
+     * limitations on how those may be used by the database. Please read the Neo4j Cypher Manual for more up-to-date
+     * details. For example, at the time of writing, it is not possible to store maps as properties
```

Comment:
```suggestion
     * In addition, please note that while this mapping allows nested structures, like map of maps, there may be
     * limitations on how those are supported by the database. Please read the Neo4j Cypher Manual for more up-to-date
     * details. For example, at the time of writing, it is not possible to store maps as properties
```
Trying to reduce the recurrence of "may be"s.